src/interfaces/chain.h

```diff
@@ -171,6 +172,9 @@ class Chain
         const uint256& ancestor_hash,
         const FoundBlock& ancestor_out={}) = 0;
 
+    //! Get a CBlock by its index and return whether successful.
+    virtual bool getBlockByIndex(CBlock& block, const CBlockIndex* pindex) = 0;
```

Comment:
a614ac59cd753c7ef83899602e0176654c5fff87, nit, naming:
```suggestion
    virtual bool getBlockByIndex(CBlock& block, const CBlockIndex* index) = 0;
```

Reply:
No, it is not wrong. Just mentioned it as in new code we do not make a reference to a type of the variable in its name:
```
git grep -e "CBlockIndex\* index"
```